lib/ruby_lsp/utils.rb

```diff
@@ -92,4 +92,36 @@ def enabled?(feature)
       @configuration[:enableAll] || @configuration[feature]
     end
   end
+
+  class HoverResponse
+    extend T::Sig
+
+    CATEGORIES = T.let([:link, :documentation, :signature], T::Array[Symbol])
+
+    sig { returns(String) }
+    attr_reader :markdown_content
+
+    sig { params(category: Symbol, markdown_content: String).void }
+    def initialize(category, markdown_content)
+      raise ArgumentError, "Invalid category: #{category}" unless CATEGORIES.include?(category)
```

Comment:
Would it be less fragile if we allow any category name, but treat it as lowest precedence if not recognized? Then an addon can introduce a new category without breaking older versions of the extension.

Reply:
I feel like we'd end up in a similar situation though. If two addons want to create their own categories, how do we determine the order?

Also, if an addon wants to put something at the bottom part, they can always just use `documentation`. I feel like if they wanted a new category it's because they would like to be above `documentation`, but then how do we know where exactly?

Reply:
I would vote for keeping it so that we have a deterministic order for the hover content. If we come to the conclusion that we missed too many categories, we can revert this decision.